pygmt/src/grdgradient.py

```diff
@@ -0,0 +1,116 @@
+"""
+grdgradient - Create a gradient from a grid.
+"""
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    args_in_kwargs,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    A="azimuth",
+    D="direction",
+    E="radiance",
+    G="outgrid",
+    R="region",
+    V="verbose",
+)
+@kwargs_to_strings(
+    A="sequence",
+    R="sequence",
+)
+def grdgradient(grid, **kwargs):
+    r"""
+    Compute the directional derivative in a given ``azimuth`` or
+    ``direction`` of the vector gradient of the data.
+
+    Full option list at :gmt-docs:`grdgradient.html`
+
+    {aliases}
+    Parameters
+    ----------
+    grid : str or xarray.DataArray
+        The file name of the input grid or the grid loaded as a DataArray.
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    azimuth : str or list or xarray.DataArray
+        *azim*\ [/*azim2*].
+        Azimuthal direction for a directional derivative; *azim* is the
+        angle in the x,y plane measured in degrees positive clockwise from
+        north (the +y direction) toward east (the +x direction). The
+        negative of the directional derivative, -[dz/dx\*sin(*azim*) +
+        dz/dy\*cos(\ *azim*)], is found; negation yields positive values
+        when the slope of z(x,y) is downhill in the *azim* direction, the
+        correct sense for shading the illumination of an image by a light
+        source above the x,y plane shining from the *azim* direction.
+        Optionally, supply two azimuths, *azim*/*azim2*, in which case the
+        gradients in each of these directions are calculated and the one
+        larger in magnitude is retained; this is useful for illuminating data
+        with two directions of lineated structures, e.g., *0*/*270*
+        illuminates from the north (top) and west (left).  Finally, if *azim*
+        is a file it must be a grid of the same domain, spacing and
+        registration as *ingrid* that will update the azimuth at each output
```

Comment:
```suggestion
        registration as *ingrid* that will update the azimuth at each output
```
Is *ingrid* used at any other position or do you mean *grid* here?

Reply:
Good catch! Done in f433363